clients/src/main/java/org/apache/kafka/common/protocol/types/Struct.java

```diff
@@ -75,6 +75,38 @@ public Object get(Field field) {
         return getFieldOrDefault(field);
     }
 
+    public Byte get(FieldDef.Int8 field) {
+        return getByte(field.name);
+    }
+
+    public Integer get(FieldDef.Int32 field) {
+        return getInt(field.name);
+    }
+
+    public Short get(FieldDef.Int16 field) {
+        return getShort(field.name);
+    }
+
+    public String get(FieldDef.Str field) {
+        return getString(field.name);
+    }
+
+    public String get(FieldDef.NullableStr field) {
+        return getString(field.name);
+    }
+
+    public Integer getOrElse(FieldDef.Int32 field, int alternative) {
+        if (hasField(field.name))
+            return getInt(field.name);
+        return alternative;
+    }
+
+    public String getOrElse(FieldDef.NullableStr field, String alternative) {
```

Comment:
A bit annoying that `getOrElse` has to be repeated for each `FieldDef`. Actually, the same for `get`. But probably fine for now. A nicer way from an abstraction perspective would be to return a `FieldValue` that one could then call `get`, `getOrElse`, etc. If `FieldDef` was generic, a single `get` method might be enough. However, there are also performance implications, so let's not pursue this option for now.

Reply:
The overloads seemed like a non-intrusive way to get better type safety. Agreed that there may still be room for improvement.